File: 2018/day01/solution.rb
require 'set'

input=File.open('input.txt').read
input.gsub!(/\r\n?/, "\n")
f=0
fs = Set[0]
solution = false
first = false
until solution do
    input.each_line do |line|
        f+=line.to_i
        if !solution & fs.include?(f) 
            puts f
            solution = f
        end
        fs.add f
    end
    if first then
        first = false
        puts f
    end
end
puts f

File: 2018/day01/solution.py
jumps = [int(line) for line in open("input.txt", "r").read().split("\n")]
f = 0
fs = {0}
found = False
while not found:
    for jump in jumps:
        f += jump
        if not found and f in fs:
            print(f)
            found = True
        fs.add(f)
print(len(fs) == len(set(fs)))
print(f)
